build/minikube_setup.sh

```diff
@@ -15,42 +15,91 @@ if ! command -v yq; then
+
+    echo "*** Installing Strimzi resources ..."
+    kubectl apply -f . -n $STRIMZI_OPERATOR_NS
+    cd "$ROOT_DIR"
+}
 
-echo "*** Unpacking .tar.gz ..."
-tar xzf strimzi-${VERSION}.tar.gz
 
-echo "Setting namespaces (OPERATOR_NS: $OPERATOR_NS, TEST_NS: $TEST_NS) in strimzi configs ..."
-cd strimzi-${VERSION}/install/cluster-operator
-# Set namespace that operator runs in
-sed -i "s/namespace: .*/namespace: ${OPERATOR_NS}/" *RoleBinding*.yaml
-# Set namespaces that operator watches
-yq eval -i "del(.spec.template.spec.containers[0].env.[] | select(.name == \"STRIMZI_NAMESPACE\").valueFrom)" 060-Deployment-strimzi-cluster-operator.yaml
-yq eval -i "(.spec.template.spec.containers[0].env.[] | select(.name == \"STRIMZI_NAMESPACE\")).value = \"$WATCH_NS\"" 060-Deployment-strimzi-cluster-operator.yaml
+function install_prometheus_operator {
+    cd "$DOWNLOAD_DIR"
+    PROM_VERSION=0.45.0
+    PROM_TARFILE="prometheus-operator-${PROM_VERSION}.tar.gz"
+    if ! test -f ${PROM_TARFILE}; then
+        echo "*** Downloading ${PROM_TARFILE} ..."
+        curl -LsS -o ${PROM_TARFILE} https://github.com/prometheus-operator/prometheus-operator/archive/v${PROM_VERSION}.tar.gz
+    fi
 
-echo "*** Creating namespaces..."
-kubectl create namespace $OPERATOR_NS || echo " ... ignoring that error"
-kubectl create namespace $TEST_NS || echo " ... ignoring that error"
+    echo "*** Unpacking .tar.gz ..."
+    tar xzf ${PROM_TARFILE}
 
-echo "*** Adding RoleBindings to namespaces watched by Strimzi ..."
-IFS=","
-for ns in $WATCH_NS; do
-    kubectl apply -f 020-RoleBinding-strimzi-cluster-operator.yaml -n $ns
-    kubectl apply -f 031-RoleBinding-strimzi-cluster-operator-entity-operator-delegation.yaml -n $ns
-    kubectl apply -f 032-RoleBinding-strimzi-cluster-operator-topic-operator-delegation.yaml -n $ns
-done
+    echo "*** Installing Prometheus operator bundle..."
+    cd prometheus-operator-${PROM_VERSION}
+    kubectl apply -f bundle.yaml
+    cd "$ROOT_DIR"
+}
 
-echo "*** Installing Strimzi resources ..."
-kubectl apply -f . -n $OPERATOR_NS
+STRIMZI_DEPLOYMENT=$(kubectl get deployment strimzi-cluster-operator -n $STRIMZI_OPERATOR_NS --ignore-not-found -o jsonpath='{.metadata.name}')
+PROM_DEPLOYMENT=$(kubectl get deployment prometheus-operator -n $PROM_OPERATOR_NS --ignore-not-found -o jsonpath='{.metadata.name}')
+
+if [ -z "$STRIMZI_DEPLOYMENT" ]; then
+    echo "*** Strimzi operator deployment not found, installing ..."
+    install_strimzi_operator
+else
+    echo "*** Strimzi operator deployment found, skipping install ..."
+fi
+
+if [ -z "$PROM_DEPLOYMENT" ]; then
+    echo "*** Prometheus operator deployment not found, installing ..."
+    install_prometheus_operator
+else
+    echo "*** Prometheus operator deployment found, skipping install ..."
+fi
 
 echo "*** Waiting for Strimzi operator to come up ..."
```

Comment:
could you background the `rollout status` commands in the respective `if` block and then call `wait` at the bottom instead?